src/index.js

```diff
@@ -1,41 +1,106 @@
+    const todosArr = storage();
+   const index = todosArr.length;
+   const description = document.getElementById('input').value;
+   const completed = false;
+   const todoList = new todo(description,completed,index);
+   if(description != ''){
+    todosArr.push(todoList);
+    localStorage.setItem('deeds', JSON.stringify(todosArr));
+   }
+}
 
-todos.forEach((deed) => {
-    const content = `
-            <div class="list">
+const Dynamic = document.querySelector('.todo-maker');
+let todos2=() => {
+    const todosArr = storage();
+    Dynamic.innerHTML = '';
+    for (let i = 0; i < todosArr.length; i += 1){
+        let isComplte = todosArr[i].completed == true ? "checked" : "";
+        Dynamic.innerHTML += `
+        <div class="list ${isComplte}">
                 <div class="list-action">
-                    <input type="checkbox">
-                    <p>${deed.description}</p>
+                    <input class ="check" id ="${todosArr[i].index}" ${isComplte} type="checkbox">
+                    <p id ="${todosArr[i].index}" contenteditable="true"class="p">${todosArr[i].description}</p>
                 </div>
                 <div class="action-container">
-                    <div><i class="fa-solid fa-ellipsis-vertical"></i></div>
+                    <div class="icon"><i id = "${todosArr[i].index}" class="fa-solid fa-ellipsis-vertical line"></i> <i id = "${todosArr[i].index}" class="fa-solid fa-trash remove"></i></div>
                 </div>
             </div>
-            <hr>`;
-            Dynamic.innerHTML += content;
-})
\ No newline at end of file
+            `;
+
+    }
+}
+
+todos2();
+
+const checked = document.querySelectorAll('.check');
+checked.forEach((button) => {
+    const todosArr = storage();
+    button.addEventListener('click', (e) => {
+       const card = e.path.filter((el) => el.classList?.contains('list')).at(0);
+       if(button.checked){
+        card.classList.add("checked");
+        todosArr[button.id].completed = true;
+       } else {
+        card.classList.remove("checked")
+        todosArr[button.id].completed = false;
+       }
+       localStorage.setItem('deeds', JSON.stringify(todosArr));
+    });
+  });
+
+document.addEventListener('keyup', (e) => {
+    const todosArr = storage();
+    if (e.key === 'Enter') {
+        localStorage.setItem('deeds', JSON.stringify(todosArr));
+         pushtodo();
+         todos2();
+         window.location.reload();
+        console.log(Dynamic)
+    }
+});
+
+const rmv = document.querySelectorAll('.remove');
+ rmv.forEach((item) => {
+  item.addEventListener('click', (e) => {
+    let todosArr = storage();
+     const card2 = e.path.filter((el) => el.classList?.contains('list')).at(0);
+     card2.remove()
+     todosArr.splice(item.id, 1);
+     todosArr.forEach((todo) => {
+        todo.index = todosArr.indexOf(todo);
+     });
+     localStorage.setItem('deeds', JSON.stringify(todosArr));
+     todos2();
+     location.reload();
+      });
+});
```

Comment:
- [ OPTIONAL] When I try to delete an item, I get this error message in the console:

![image](https://user-images.githubusercontent.com/77106962/206549602-9e6af973-82e7-4561-8f3c-019a77e74be8.png)

I would advise you to delete these two lines of code:

```JavaScript
 const card2 = e.path.filter((el) => el.classList?.contains('list')).at(0);
 card2.remove()
```

Reply:
@quavo19,  Kindly check, I can't edit your tasks. On `keyup`, they do nothing. Please check again. The recommendation I gave, worked and the tasks were edited. 😄

Reply:
@Gambit142 

https://www.loom.com/share/b41384ffb5fb4e1b98b609abf7a67987

Reply:
@Gambit142 
everything works fine

Reply:
@Gambit142 works in incognito mode as well

Reply:
@quavo19, this is what I see at my end

https://www.loom.com/share/d524cec242c247768844301f12a4273f

#### If you feel I didn't give you the correct review, kindly ask for a second opinion by clicking this [form](https://airtable.com/shrQAqnBwek5a0O0s)

Reply:
@Gambit142 I've been able to work on the required changes